### ISSUE TITLE
[D][A11Y] CKEditor: Add the "Language" (text part language) button as an option

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6516]
Our telemetry currently shows 40% of Backdrop sites are using either just v4 or both v4 and v5 of CKEditor, so this feature should be implemented to both versions.

- CKE v4 addon: https://ckeditor.com/cke4/addon/language
  > This plugin implements the Language toolbar button to support the [WCAG 3.1.2 Language of Parts](http://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-other-lang-id.html) specification.

  ![image](https://github.com/backdrop/backdrop-issues/assets/2423362/29950f03-eaef-4bda-9fdc-183cb0be9bde)
- CKE v5 feature: https://ckeditor.com/docs/ckeditor5/latest/features/language.html
  > The text part language feature lets you mark the language of text fragments. This way browsers and screen readers can correctly interpret parts written in different languages.
  <img width="781" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/d01b2f4c-3c78-4c1d-91df-f71caf5ad0ed">

In Drupal, the button is available for the CKE toolbar in the text format configuration form:
<img width="984" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/2a4a3580-85d9-4069-bea8-5fb34823f079">

...adding it to the active toolbar reveals a set of options that allow configuring which languages should be available in the toolbar button when editing content:
<img width="950" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/e2ad7749-1244-4bb8-a237-547ea5cdde9d">

With https://www.drupal.org/project/drupal/issues/3273986, the 3rd "Site configured" option was added, which automatically lists all languages that have been enabled/added to the site.

**Comment by klonos** [comment]
...since people can add text in any language when creating/editing content, we should not be necessarily restricting the languages to those installed on the site, but I think that it does make sense to make that the default option, because:
- the set of languages in the "6 UN official languages" option ([Arabic, Chinese, English, French, Russian and Spanish](https://www.un.org/en/our-work/official-languages)) does not make sense for all sites.
- showing all languages makes it a bad UX, as it is difficult to find a language in a set of 100+ languages in a single select.

We should also make that option act "dynamically": when the "Site-configured languages" option is selected for a format, then the drop-down menu in CKE should be automatically updating accordingly to reflect the set of languages in `admin/config/regional/language`.
